Add 8.8 Announcements (#4888)
camunda/camunda-docs: docs/reference/announcements/880.md

For SaaS customers, identity management in Camunda 8.7 remains consistent with Camunda 8.6, allowing the attachment of a single IdP per organization. However, cluster-level identity capabilities are provided for SaaS as well as Self-Managed. This means that user groups, roles, and access permissions can now be managed at the cluster level, giving SaaS customers the same granular access control as in Self-Managed environments.

### Decoupling from Keycloak <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>

Built-in Keycloak integration in Self-Managed is removed, allowing customers to use any compatible IdP.

- Keycloak remains fully supported as an external option. For cluster-level identity management it must be connected as an external OIDC provider moving forward.
- OpenID Connect (OIDC) remains the standard for seamless integration with chosen IdPs.

### Resource-based permissions

> Resource-level permissions are introduced for process definitions and web applications.

Reviewer: ```suggestion
Resource-level permissions are introduced to control read and write permissions per specific resource.
```

- Admin users retain full access, but regular users must be granted specific permissions to perform actions/view resources.

> - For organizations that build custom front-ends and access Camunda via API, users with API permissions can still access process data through the V2 API.

Reviewer: ```suggestion
- For organizations that build custom front-ends and access Camunda via API, users or Clients with API permissions can still access data through the V2 API, respecting their resource permissions.
```

<!-- :::info
Learn more about these updates in Introducing Enhanced Identity Management in Camunda 8.7.
::: -->

## Installation and deployment updates <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>

> Camunda 8.7 introduces a streamlined architecture, consolidating core components such as Zeebe, Operate, and Tasklist into a single deployable unit. Enhanced deployment options are also included, such as new Kubernetes Helm guides, deployment reference architectures, and improved support for professional developers with Camunda 8 Run.

Reviewer: ```suggestion
Camunda 8.8 introduces a streamlined architecture, consolidating core components such as Zeebe, Operate, Tasklist, Optmize, and Connectors into a single deployable unit. Enhanced deployment options are also included, such as new Kubernetes Helm guides, deployment reference architectures, and improved support for local development with Camunda 8 Run.
```

You can download the alpha release of the unified package from the Camunda GitHub repository, either as an executable Java application (Camunda Orchestration Core) or a Docker image.

### Helm charts

> If you are using the recommended Camunda 8 deployment option (Helm charts), the upgrade path from version 8.6 to 8.7 will be straightforward by changing the values file to the new syntax.

Reviewer: ```suggestion
If you are using the recommended Camunda 8 deployment option (Helm charts), the upgrade path from version 8.7 to 8.8 will be straightforward by changing the values file to the new syntax.
```

New migration guides will also be provided to support you when migrating from a previous Camunda version.

:::caution

> Additional upgrade considerations are necessary for deployments that use custom scripts, such as Docker containers, manual installations, or custom-developed Kubernetes deployments. For these deployments, customers can either continue to deploy with their original 8.6 topology and upgrade each component independently, or adopt our Helm Chart approach for the upgrade, which allows for unifying the deployment into a single JAR or container executable.

Reviewer: ```suggestion
Additional upgrade considerations are necessary for deployments that use custom scripts, such as Docker containers, manual installations, or custom-developed Kubernetes deployments. For these deployments, customers can either continue to deploy with their original 8.7 topology and upgrade each component independently, or adopt our Helm Chart approach for the upgrade, which allows for unifying the deployment into a single JAR or container executable.
```

The following diagram shows a simplified version of this work.

![Camunda Exporter diagram](../img/target-camunda-exporter.png)

- For example, Tasklist and Operate Importers are still required for old data to be imported, but the Camunda exporter writes all new data into ES/OS. After old indices are drained, importers can be turned off.
- The archiver, which takes care of the archiving of completed process instances, will be moved into the Zeebe system as well, to reduce the installation complexity and provide a better scaling and replication factor (based on partitions).
- This helps achieve a streamlined architecture, and improves platform performance and stability (especially regarding ES/OS).
- A new separate component covers the migration, which will be part of the single application but can also deployed separately. It will adjust the previous Operate indices to make them more harmonized and usable by Tasklist.

<!-- :::info
Learn more about these updates in Streamlined Deployment with 8.7.
::: -->

## Camunda Java client and Camunda Spring SDK <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>

> With the Camunda 8.8 release, Camunda Java client and Camunda Spring SDK replace the Zeebe Java client and Zeebe Spring SDK. This allows you to use a single consolidated client to interact with Camunda clusters.

Reviewer: ```suggestion
With the Camunda 8.8 release, Camunda Java Client and Camunda Spring SDK replace the Zeebe Java client and Zeebe Spring SDK. This allows you to use a single consolidated client to interact with Camunda orchestration clusters.
```

> The `CamundaClient` replaces the `ZeebeClient`, offering the same functionality and adding new capabilities.

Reviewer 2: ```suggestion
The `CamundaClient` replaces the `ZeebeClient`, offering the same functionality and adding new capabilities. You can use the new `io.camunda:camunda-client-java` artifact to receive the new `CamundaClient`.
```

:::note

> - If you need to continue using the old `ZeebeClient`, you can use the version 8.6 artifact without any issues with newer cluster versions as the client is forward-compatible.

Reviewer: ```suggestion
- If you need to continue using the old `ZeebeClient`, you can use the version 8.7 artifact without any issues with newer cluster versions as the client is forward-compatible.
```

Reviewer: @tmetzke, is that something that we can guarantee?

Author: @aleksander-dytko I committed all your suggestions, but left this one for @tmetzke to chime in on.

Author: @tmetzke can you confirm this is correct for 8.8?

>If you need to continue using the old `ZeebeClient`, you can use the version 8.7 artifact without any issues with newer cluster versions as the client is forward-compatible.

Reviewer 2: With 8.8, you'll still have the ZeebeClient classes in parallel to the CamundaClient classes. There is a new artifact with 8.8, `camunda-client-java`, that can be used as a drop-in replacement for `zeebe-client-java` as it contains the Zeebe client classes as well (marked as deprecated so you can see in your code that you should move away from them).

With 8.8, the `zeebe-java-client` artifact becomes a so-called relocation artifact, containing a relocation/redirect instruction for Maven to use the `camunda-client-java` artifact instead. Maven processors should issue a warning in the logs for this as well when a project is built using the old artifact.

With 8.9, the `zeebe-client-java` artifact will not be distributed anymore. Also, the `camunda-java-client` artifact will not contain Zeebe client classes anymore with 8.9.

Reviewer 2: There is an open docs issue on our end to clear this up: https://github.com/camunda/camunda-docs/issues/4046

Author: @tmetzke so we need to adjust this announcement to mention the 8.8 artifact, not the 8.7 one?

Author: Without confirmation, I'm blocked from merging this. Please respond ASAP @tmetzke and @aleksander-dytko so we can make sure this is correct tomorrow for the release.

Reviewer 2: ```suggestion
- If you need to continue using the old `ZeebeClient`, you can use the new version 8.8 `CamundaClient` artifact without issues as it still contains the related `ZeebeClient` classes. Those classes are marked as deprecated, so you can easily spot code you need to adjust to the `CamundaClient`.
- The old `zeebe-client-java` artifact is now relocation-only, so your build system is redirected to the new `camunda-client-java` artifact. We will discontinue the old artifact with a future release.
```

Reviewer 2: @akeller, I see this is marked as resolved but the content is still the previous one mentioning 8.6. What is the current state of this? We start to have follow-up PRs like [this one](https://github.com/camunda/camunda-docs/pull/5049) that build on top of text that was supposed to be fixed already, as far as I'm concerned.

Author: Sounds like it simply got lost. @christinaausley can you follow up on this with @tmetzke and team?

> - The Zeebe Java client will not be developed further and only receives bug fixes while version 8.6 is officially supported.

Reviewer: ```suggestion
- The Zeebe Java client will not be developed further and only receives bug fixes while version 8.7 is officially supported.
```

:::

### Key changes

| Change                                          | Description                                                                                                                                                                                                                                                                                                        |
| :---------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | New package structure                           | Package `io.camunda.client`: Contains the new `CamundaClient` and all 8.7 features.                                                                                                                                                                                                                                |

Reviewer 2: ```suggestion
| New package structure                           | Package `io.camunda.client`: Contains the new `CamundaClient` and all 8.8 features.                                                                                                                                                                                                                                |
```

---
id: announcements-880
title: "8.8 Announcements"
description: "Important changes and updates for the Camunda 8.8 release including deprecation & removal notices."
---

import DeployDiagramImg from '../img/deploy-diagram-modal.png';

Important changes and updates for the Camunda 8.8 release are summarized below.

| Scheduled release date | Scheduled end of maintenance | Release notes | Blog |
| :--------------------- | :--------------------------- | :------------ | :--- |

> | 14 October 2025        | 14 April 2026                |               |      |

Reviewer 3: ```suggestion
| 14 October 2025        | 13 April 2027                |               |      |
```

### Decoupling from Keycloak <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>

Built-in Keycloak integration in Self-Managed is removed, allowing customers to use any compatible IdP.

- Keycloak remains fully supported as an external option. For cluster-level identity management it must be connected as an external OIDC provider moving forward.
- OpenID Connect (OIDC) remains the standard for seamless integration with chosen IdPs.

### Resource-based permissions

Resource-level permissions are introduced to control read and write permissions per specific resource.

- Admin users retain full access, but regular users must be granted specific permissions to perform operations or view resources.
- For organizations that build custom front-ends and access Camunda via API, users or Clients with API permissions can still access data through the V2 API, respecting their resource permissions.

<!-- :::info
Learn more about these updates in Introducing Enhanced Identity Management in Camunda 8.7.
::: -->

## Installation and deployment updates <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>

> Camunda 8.8 introduces a streamlined architecture, consolidating core components such as Zeebe, Operate, Tasklist, Optmize, and Connectors into a single deployable unit. Enhanced deployment options are also included, such as new Kubernetes Helm guides, deployment reference architectures, and improved support for local development with Camunda 8 Run.

Reviewer 4: Can we deep-link the Helm guides, reference architecture and C8Run?

Author: @conceptualshark the link to C8Run is obvious, but are the others mostly on the install page? What would you link to?

Author: @conceptualshark I'll add the reference arch page, but please follow up in a new PR when other adjustments are made and can be linked.

New migration guides will also be provided to support you when migrating from a previous Camunda version.

:::caution
Additional upgrade considerations are necessary for deployments that use custom scripts, such as Docker containers, manual installations, or custom-developed Kubernetes deployments. For these deployments, customers can either continue to deploy with their original 8.7 topology and upgrade each component independently, or adopt our Helm chart approach for the upgrade, which allows for unifying the deployment into a single JAR or container executable.
:::

### Camunda Exporter

A new Camunda Exporter brings the importer and archiving logic of web components (Tasklist and Operate) closer to the distributed platform (Zeebe). The index schema is also being harmonized.

#### Harmonized index schema

Camunda is harmonizing our index structure and usage.

- This removes unnecessary duplications over multiple indices due to the previous architecture.
- With this change, several Operate indices can and will be used by Tasklist.
- New indices have been created to integrate Identity into the system.

![Harmonized indices schema](../img/harmonized-indices-schema.png)

> #### Camunda Exporter

Author: ❓ See line 105. Maybe that's the one you want to bring to an H2?

> #### Camunda Exporter

Reviewer 4: Ah yes - sorry. line 105 is what my intention was to bring to H2 :-)

The following diagram shows a simplified version of this work.

![Camunda Exporter diagram](../img/target-camunda-exporter.png)

- For example, Tasklist and Operate Importers are still required for old data to be imported, but the Camunda exporter writes all new data into ES/OS. After old indices are drained, importers can be turned off.
- The archiver, which takes care of the archiving of completed process instances, will be moved into the Zeebe system as well, to reduce the installation complexity and provide a better scaling and replication factor (based on partitions).
- This helps achieve a streamlined architecture, and improves platform performance and stability (especially regarding ES/OS).
- A new separate component covers the migration, which will be part of the single application but can also deployed separately. It will adjust the previous Operate indices to make them more harmonized and usable by Tasklist.

<!-- :::info
Learn more about these updates in Streamlined Deployment with 8.7.
::: -->

> ## Camunda Java client and Camunda Spring SDK <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>

Reviewer 4: The Java Client and SDK are also relevant for SaaS use-cases, or not?

```suggestion
## Camunda Java client and Camunda Spring SDK
```

Resource-level permissions are introduced for process definitions and web applications.

- Admin users retain full access, but regular users must be granted specific permissions to perform actions/view resources.
- For organizations that build custom front-ends and access Camunda via API, users with API permissions can still access process data through the V2 API.

<!-- :::info
Learn more about these updates in Introducing Enhanced Identity Management in Camunda 8.7.
::: -->

## Installation and deployment updates <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>

Camunda 8.7 introduces a streamlined architecture, consolidating core components such as Zeebe, Operate, and Tasklist into a single deployable unit. Enhanced deployment options are also included, such as new Kubernetes Helm guides, deployment reference architectures, and improved support for professional developers with Camunda 8 Run.

You can download the alpha release of the unified package from the Camunda GitHub repository, either as an executable Java application (Camunda Orchestration Core) or a Docker image.

:::caution breaking change: Deploy diagram modal

> The Web Modeler **Deploy diagram** modal has changed, and clusters must now be proactively configured to be able to deploy from Web Modeler.

Reviewer 4: This should move to 8.7

Author: Done
